File: csnippets/insertion_sort.c
#include <stdio.h>

int main(void) {
	int arr[]={5,25,14,16,12,11,2,3,4,77};
	int n=sizeof(arr)/sizeof(arr[0]);
	int j=0;
	int temp=0;
	for(int i=1;i<n;i++){
	    temp=arr[i];
	    j=i-1;
	    while(j>=0 && arr[j]>temp){
	        arr[j+1]=arr[j];
	        j--;
	    }
	    arr[j+1]=temp;
	}
	printf("Sorted Array: ");
	for(int k=0;k<n;k++){
	    printf("%d ,",arr[k]);
	}
	return 0;
}

File: pythonsnippets/insertion_sort.py
arr=[5,6,54,8,65,67,56,97,2,47]
for i in range(1,len(arr)):
    j=i-1
    temp=arr[i]
    while j>=0 and arr[j]>temp:
        arr[j+1]=arr[j]
        j=j-1
    arr[j+1]=temp
for n in range(0,len(arr)-1):
    print(arr[n], end =" ")
